setup.cfg

```diff
@@ -1,3 +1,30 @@
+[metadata]
+name = pymemcache
+version = attr: pymemcache.__version__
```

Comment:
Will this approach run into the same issues as #214 (fixed by #216)? Or can we guarantee that `six` (from `install_requires`) has already been installed when this attribute is evaluated?

Reply:
Good catch!!!!

```shell
hberaud@localhost$ git clone git@github.com:4383/pymemcache -b packaging                                                                                                                                    
Clonage dans 'pymemcache'...                                                                                                                                                                                         
remote: Enumerating objects: 4, done.                                                                                                                                                                                
remote: Counting objects: 100% (4/4), done.                                                                                                                                                                          
remote: Compressing objects: 100% (4/4), done.                                                                                                                                                                       
remote: Total 1914 (delta 0), reused 3 (delta 0), pack-reused 1910                                                                                                                                                   
Réception d'objets: 100% (1914/1914), 404.50 KiB | 784.00 KiB/s, fait.                                                                                                                                               
Résolution des deltas: 100% (1213/1213), fait.                                                                                                                                                                       
hberaud@localhost$ cd pymemcache     
hberaud@localhost$ pipenv shell
Creating a virtualenv for this project…
Pipfile: /tmp/pymemcache/Pipfile
Using /usr/bin/python3.7 (3.7.2) to create virtualenv…
⠇ Creating virtual environment...Already using interpreter /usr/bin/python3.7
Using base prefix '/usr'
New python executable in /home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/bin/python3.7
Also creating executable in /home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/bin/python
Installing setuptools, pip, wheel...
done.

✔ Successfully created virtual environment!
Virtualenv location: /home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8
Creating a Pipfile for this project…
Launching subshell in virtual environment…
 . /home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/bin/activate
 hberaud@localhost  /tmp/pymemcache   packaging   . /home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/bin/activate
(pymemcache)  hberaud@localhost$ pip freeze
(pymemcache)  hberaud@localhost$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    setup()
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib64/python3.7/distutils/core.py", line 121, in setup
    dist.parse_config_files()
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/dist.py", line 705, in parse_config_files
    ignore_option_errors=ignore_option_errors)
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 120, in parse_configuration
    meta.parse()
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 425, in parse
    section_parser_method(section_options)
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 398, in parse_section
    self[name] = value
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 183, in __setitem__
    value = parser(value)
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 513, in _parse_version
    version = self._parse_attr(value, self.package_dir)
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib/python3.7/site-packages/setuptools/config.py", line 348, in _parse_attr
    module = import_module(module_name)
  File "/home/hberaud/.local/share/virtualenvs/pymemcache-J0J5VsH8/lib64/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/pymemcache/pymemcache/__init__.py", line 3, in <module>
    from pymemcache.client.base import Client  # noqa
  File "/tmp/pymemcache/pymemcache/client/__init__.py", line 3, in <module>
    from pymemcache.client.base import Client  # noqa
  File "/tmp/pymemcache/pymemcache/client/base.py", line 16, in <module>
    import six
ModuleNotFoundError: No module named 'six'
```

Reply:
I guess simply use a `setup_requires` can avoid all these kinds of issues due to dependencies loading/ordering... let's me do a test :)

Reply:
Yeah that works!!!!!!